src/components/WorkspaceMenu.js

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from 'react';
+import { compose } from 'redux';
+import Menu from '@material-ui/core/Menu';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import MenuItem from '@material-ui/core/MenuItem';
+import Typography from '@material-ui/core/Typography';
+import ViewHeadlineIcon from '@material-ui/icons/ViewHeadline';
+import { withStyles } from '@material-ui/core/styles';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import ConnectedWindowList from './WindowList';
+
+/**
+ */
+export class WorkspaceMenu extends Component {
+  /**
+   * constructor -
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      windowListAnchorEl: null,
+    };
+    this.handleWindowListClick = this.handleWindowListClick.bind(this);
+    this.handleWindowListClose = this.handleWindowListClose.bind(this);
+  }
+
+  /**
+   * @private
+   */
+  handleWindowListClick(event) {
+    this.setState({
+      windowListAnchorEl: event.currentTarget,
+    });
+  }
+
+  /**
+   * @private
+   */
+  handleWindowListClose() {
+    this.setState({
+      windowListAnchorEl: null,
+    });
+  }
+
+  /**
+   * render
+   * @return
+   */
+  render() {
+    const { handleClose, anchorEl } = this.props;
+    const { windowListAnchorEl } = this.state;
+
+    return (
+      <>
+        <Menu id="workspace-menu" anchorEl={anchorEl} open={Boolean(anchorEl)} onClose={handleClose}>
+          <MenuItem
+            aria-haspopup="true"
+            onClick={(e) => { this.handleWindowListClick(e); handleClose(e); }}
```

Comment:
@jvine @ggeisler: I'm not sure what the intended interaction pattern for working with these submenus is; this PR just closes the parent menu and leaves the child menu, and I'm hoping we can get away with that for the near future (or, I suspect you ultimately want some controls from the child menu to close both, and some controls to only close the child menu... it's probably doable, but ugly, and would be nice to tackle after we have a better established pattern..)

Reply:
Not sure if I'm understanding this correctly, but I think the pattern should be that the parent menu is closed when a child menu/dialog/whatever is opened. Is that what you mean by "closes the parent menu and leaves the child menu"?

Reply:
Perfect; I was worried there was some statefulness implied by the close button.